src/status_im/chat/models/delete_message_for_me.cljs

```diff
@@ -0,0 +1,69 @@
+(ns status-im.chat.models.delete-message-for-me
+  (:require [re-frame.core :as re-frame]
+            [status-im.ethereum.json-rpc :as json-rpc]
+            [status-im.utils.fx :as fx]
+            [taoensso.timbre :as log]))
+
+(defn- update-db-clear-delete-message-for-me-undo-timer
+  [db chat-id message-id]
+  (when (get-in db [:messages chat-id message-id])
+    (update-in db
+               [:messages chat-id message-id]
+               dissoc
+               :deleted-for-me-undoable-till)))
+
+(defn- update-db-delete-message-for-me-locally
+  [db chat-id message-id undo-time-limit-ms]
+  (when (get-in db [:messages chat-id message-id])
+    (update-in db
+               [:messages chat-id message-id]
+               assoc
+               :deleted-for-me? true
+               :deleted-for-me-undoable-till (+ (js/Date.now)
+                                                undo-time-limit-ms))))
+
+(defn- update-db-undo-delete-message-for-me-locally
+  [db chat-id message-id]
+  (let [{:keys [deleted-for-me? deleted-for-me-undoable-till]}
+        (get-in db [:messages chat-id message-id])]
+    (if (and deleted-for-me? (> deleted-for-me-undoable-till (js/Date.now)))
+      (update-in db
+                 [:messages chat-id message-id]
+                 dissoc
+                 :deleted-for-me?
+                 :deleted-for-me-undoable-till)
+      (update-db-clear-delete-message-for-me-undo-timer db
+                                                        chat-id
+                                                        message-id))))
+
+(fx/defn delete-message-for-me
+  "Delete message for me now locally and broadcast after undo time limit timeout"
+  {:events [:chat.ui/delete-message-for-me]}
+  [{:keys [db]} {:keys [chat-id message-id]} undo-time-limit-ms]
+  (when (get-in db [:messages chat-id message-id])
+    {:db (update-db-delete-message-for-me-locally db chat-id message-id undo-time-limit-ms)
+
+     :utils/dispatch-later
+     [{:dispatch [:chat.ui/delete-message-for-me-and-sync chat-id message-id]
+       :ms       undo-time-limit-ms}]}))
+
+(fx/defn undo-delete-message-for-me
+  "Undo delete for me, only allowed in limited time"
```

Comment:
I didn't quite understand this docstring. I know what you mean, but only because I looked into the source of `update-db-undo-delete-message-for-me-locally`. I would suggest rewording it or removing it. I found the implementation to be clear already without the docstring.

Reply:
removed the doc string, add new doc string to `update-db-undo-locally`